homeassistant/components/google_assistant/trait.py

```diff
@@ -1132,6 +1119,92 @@ def query_attributes(self):
                     'Setting a position is not supported')
 
 
+@register_trait
+class VolumeTrait(_Trait):
+    """Trait to control brightness of a device.
+
+    https://developers.google.com/actions/smarthome/traits/volume
+    """
+
+    name = TRAIT_VOLUME
+    commands = [
+        COMMAND_SET_VOLUME,
+        COMMAND_VOLUME_RELATIVE,
+    ]
+
+    @staticmethod
+    def supported(domain, features, device_class):
+        """Test if state is supported."""
+        if domain == media_player.DOMAIN:
+            return ((features & media_player.SUPPORT_VOLUME_SET) and
+                    (features & media_player.SUPPORT_VOLUME_MUTE))
```

Comment:
Why is this required?

Reply:
Because it uses mute if mute is requested. I could assume mute work but I think I risk exceptions then.

Reply:
If mute not supported, we can just set volume to 0 ?

Reply:
yes, but then i need to check features in the execute method. Seems volume 0 is what google sets when asking for mute. 

It does separate the states in the reporting thou, so i don't want to forget previous volume for unmute.

Supporting the case of missing mute makes the code more complicated (it's already messier than i wanted to).

Reply:
I dropped the explicit mute support. Since this is unofficial traits, might as well keep them simple for now. It might change. So we don't need mute on the media player now. It will just set volume to 0 if requested to mute.

We might want base component to translate that to mute, but that is different issue.